src/barrier-picker/SpreadBarrierCard.js

```diff
@@ -42,7 +42,7 @@ export default class SpreadBarrierCard extends Component {
                 <InputGroup
                     type="number"
                     label={`Amount per point (${currency})`}
-                    value={spreadInfo.amountPerPoint}
+                    defaultValue={spreadInfo.amountPerPoint}
```

Comment:
I have seen all those. The value have to changed before they are fired right ? How can the value change when you fix the value by setting the value ?  Check again .